### PR TITLE
Add a bit of documentation to MPI::compute_point_to_point_communication_pattern()

### PR DESCRIPTION
**Pull request by bangerth**
Follow-up to #13097.

I took the liberty to simplify the condition at the beginning a little bit by way of a lambda function. The rest of the patch might be hard to read because I removed a gratuitous pair of `{...}`, so look at it with whitespace off.

/rebuild

### CODE REVIEW

File: source/base/mpi.cc [hunk 1]
```diff
@@ -386,136 +386,157 @@ namespace Utilities
...
           ConsensusAlgorithms::AnonymousProcess<char, char> process(
             [&]() { return destinations; });
           ConsensusAlgorithms::NBX<char, char> consensus_algorithm(process,
                                                                    mpi_comm);
           return consensus_algorithm.run();
         }
+      else
+        // If that was not the case, we need to use the remainder of the code
+        // below, i.e., just fall through the else branch here
+        {}
```
Comment (masterleinad) [inline source/base/mpi.cc:417]: I would prefer to just omit this.
```suggestion
```

**Comment by bangerth** [comment]
Updated.

**Comment by bangerth** [comment]
Ping?

**Review by masterleinad — APPROVED** [review]

File: source/base/mpi.cc (hunk 1 above)
Comment (tamiko) [reply source/base/mpi.cc]: Suggested changes spanning multiple lines, finally! :smiley:

**Pull request merged**